recipes/libx265/all/conanfile.py

```diff
@@ -45,6 +48,9 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
+        if self.settings.os != "Windows":
+            if self.options.assembly and (self.options.shared or self.options.get_safe("fPIC", True)):
+                raise ConanInvalidConfiguration("Cannot enable assembly for when building libx265 as a shared library or with fPIC enabled")
```

Comment:
I don't understand. What's the issue with shared build? Assembly was enabled (default value AFAIK), and there was no issue with shared libs on Linux and Macos.

Reply:
c3i did build libx265 with assembly disabled because it does not have nasm installed (the default depends on whether nasm is found).
I knew that something was up because a shared libx265 on Linux worked before on my system.
I traced the problem to a link flag for UNIX, which is added now to `self.cpp_info.sharedlinkflags`.

It it for Linux only for now.
The tests will probably fail for macos. Let's see.